force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,64 @@
+trigger OpportunityTrigger on Opportunity (before update, before delete) {
+
+    // Handle 'before update' context for Opportunity Amount validation
+    if (Trigger.isUpdate) {
+        for (Opportunity oppBefore : Trigger.new) {
+            if (oppBefore.Amount != null && oppBefore.Amount <= 5000) {
+                oppBefore.addError('Opportunity amount must be greater than 5000');
+            }
```

Comment:
Good job checking for null, you are the first I have seen that has done that. This might be something you need clarity in the requirements. Can amount be blank?